### DISCUSSION TITLE
queue specific metrics for Prometheus

### DISCUSSION THREAD

**Discussion by curuvija** [rabbitmq/rabbitmq-server#3468] [Other] [answered]
Would it be possible to extend Prometheus metrics to include queue specific metrics. Something like these:

Rabbitmq dead letter queue filling up - Dead letter queue is filling up (> 1 msg) (mandatory)
Rabbitmq too many messages in queue - Queue is filling up (> 1000 msgs) (mandatory)
Rabbitmq slow queue consuming - Queue messages are consumed slowly (> 60s) (mandatory)

Currently for these we are using this exporter https://github.com/kbudde/rabbitmq_exporter. It would be nice to have this out of the box and not use another exporter. I have to say I am not a programmer and I cannot contribute much on this topic besides installation and testing if that helps at all.

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment, marked as answer]
See [Aggregated vs. Per-object Metrics](https://www.rabbitmq.com/prometheus.html#metric-aggregation). Per-object metrics is not an option beyond a certain number of entities since the cardinality is too high to visualise.

What you are asking here is alerting, which is a separate concern from visualising. The plugin already allows for per-object metrics via a dedicated endpoint, `GET /metrics/per-object`. There may be more features coming out in this area in the commercial Tanzu RabbitMQ edition.

**Reply by curuvija** [reply]
@michaelklishin thanks a lot. I think I can work with that :)

**Reply by curuvija** [reply]
Hi @michaelklishin do you have any recommendations on how many queues/connections are too much to handle by the system before it chokes? Thanks.
